### PR TITLE
chore: Replace multierr with errors.Join where applicable

### PR DESCRIPTION
**Pull request by cpheps**
### Proposed Change
- Replaced `multierr` and `multierrors` with `errors.Join` where applicable
  -  I did not replace them in `model/validation/errors.go` as the custom warning formatting seemed like a good use of multierror to format
- Added a new function `cli.FormatError` to format cli output like multierr so it would be uniform regardless of how errors are created.

##### Checklist
- [x] Changes are tested
- [ ] CI has passed

### CODE REVIEW

File: client/bindplane.go [hunk 1]
```diff
@@ -654,8 +653,8 @@ func (c *bindplaneClient) CopyConfig(_ context.Context, name, copyName string) e
 	case http.StatusConflict:
 		return fmt.Errorf("a configuration with name '%s' already exists", copyName)
 	default:
-		err := &multierror.Error{}
-		multierror.Append(err, fmt.Errorf("failed to copy configuration, got status %v", resp.StatusCode()))
+		var errs error
+		errs = errors.Join(errs, fmt.Errorf("failed to copy configuration, got status %v", resp.StatusCode()))
```
Comment (andykellr) [inline client/bindplane.go:657]:
```suggestion
		errs := fmt.Errorf("failed to copy configuration, got status %v", resp.StatusCode())
```

File: client/bindplane.go [hunk 2]
```diff
@@ -665,11 +664,11 @@ func (c *bindplaneClient) CopyConfig(_ context.Context, name, copyName string) e
 
 		if len(errResponse.Errors) > 0 {
 			for _, e := range errResponse.Errors {
-				multierror.Append(err, errors.New(e))
+				errors.Join(err, errors.New(e))
```
Comment (andykellr) [inline client/bindplane.go:667]:
```suggestion
				errs = errors.Join(errs, errors.New(e))
```

**Review by andykellr — CHANGES_REQUESTED** [review]
Small tweak

File: client/bindplane.go (hunk 2 above)
Comment (cpheps) [reply client/bindplane.go:667]: Oh good catch. I'm going to do a quick search to make sure I didn't miss any more.

**Review by andykellr — APPROVED** [review]

**Review by Mrod1598 — APPROVED** [review]

**Pull request merged**